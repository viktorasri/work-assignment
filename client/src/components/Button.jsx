import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ children, variant, ...props }) => {
  return (
    <button className={`button ${variant ? `button--${variant}` : ''}`} {...props}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
  variant: PropTypes.oneOf(['primary', 'icon']),
};

export default Button;
