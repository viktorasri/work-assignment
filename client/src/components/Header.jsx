import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../assets/logo.svg';

const Header = () => {
  return (
    <header className='header'>
      <div className='row'>
        <div className='header__inner'>
          <Link to='/'>
            <img className='header__logo' src={logo} alt='logo' />
          </Link>
        </div>
      </div>
    </header>
  );
};

export default Header;
