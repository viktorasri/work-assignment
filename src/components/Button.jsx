import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ children, variant, ...props }) => {
  return (
    <button className={`button ${variant ? `button--${variant}` : ''}`} {...props}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.string]).isRequired,
  variant: PropTypes.oneOf(['primary', 'secondary', 'icon']),
};

export default Button;
