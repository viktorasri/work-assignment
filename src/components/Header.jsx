import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';

import logo from '../assets/logo.svg';
import Button from './Button';
import { logout } from '../redux/actions/userActions';

const Header = () => {
  const dispatch = useDispatch();

  const userLogin = useSelector((state) => state.userLogin);
  const { success } = userLogin;

  const logoutHandler = () => {
    dispatch(logout());
  };

  return (
    <header className='header'>
      <div className='row'>
        <div className='header__inner'>
          <Link to='/'>
            <img className='header__logo' src={logo} alt='logo' />
          </Link>
          {success && (
            <Button onClick={logoutHandler} variant='secondary'>
              Logout
            </Button>
          )}
        </div>
      </div>
    </header>
  );
};

export default Header;
