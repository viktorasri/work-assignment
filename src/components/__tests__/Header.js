import React from 'react';
import { shallow } from 'enzyme';
import * as redux from 'react-redux';
import Header from '../Header';
import Button from '../Button';

describe('<Header />', () => {
  let wrapper;
  const useSelectorSpy = jest.spyOn(redux, 'useSelector');
  const useDispatchSpy = jest.spyOn(redux, 'useDispatch');

  it('expect to render component for not signed in user', () => {
    useSelectorSpy.mockReturnValue({});
    expect(shallow(<Header />)).toMatchSnapshot();
  });

  it('expect to render component for signed user', () => {
    useSelectorSpy.mockReturnValue({ success: true });
    expect(shallow(<Header />)).toMatchSnapshot();
  });

  it('expect to render logout button for signed in user', () => {
    useSelectorSpy.mockReturnValue({ success: true });
    expect(
      shallow(<Header />)
        .find(Button)
        .exists()
    ).toBe(true);
  });

  it('should call logout when button is clicked', () => {
    const mockDispatch = jest.fn();
    useDispatchSpy.mockReturnValue(mockDispatch);
    useSelectorSpy.mockReturnValue({ success: true });

    shallow(<Header />)
      .find(Button)
      .simulate('click');

    expect(mockDispatch).toHaveBeenCalledTimes(1);

    useDispatchSpy.mockClear();
  });
});
