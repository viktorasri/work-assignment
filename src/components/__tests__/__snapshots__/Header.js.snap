// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> expect to render component for not signed in user 1`] = `
<header
  className="header"
>
  <div
    className="row"
  >
    <div
      className="header__inner"
    >
      <Link
        to="/"
      >
        <img
          alt="logo"
          className="header__logo"
          src="logo.svg"
        />
      </Link>
    </div>
  </div>
</header>
`;

exports[`<Header /> expect to render component for signed user 1`] = `
<header
  className="header"
>
  <div
    className="row"
  >
    <div
      className="header__inner"
    >
      <Link
        to="/"
      >
        <img
          alt="logo"
          className="header__logo"
          src="logo.svg"
        />
      </Link>
      <Button
        onClick={[Function]}
        variant="secondary"
      >
        Logout
      </Button>
    </div>
  </div>
</header>
`;
